File: crypto-arbitrage-platform/web_frontend/src/api/blacklist.ts
import axios from 'axios'

export interface BlacklistItem {
  id: string
  symbol: string
  exchange: string
  reason: string | null
  created_at: string
  created_by: string
}

export const blacklistApi = {
  list: () => axios.get<{ items: BlacklistItem[]; total: number }>('/api/v1/blacklist'),
  create: (data: { symbol: string; exchange: string; reason?: string }) =>
    axios.post('/api/v1/blacklist', data),
  delete: (id: string) => axios.delete(`/api/v1/blacklist/${id}`)
}

File: crypto-arbitrage-platform/web_backend/src/models/blacklist.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BlacklistItemResponse {
    pub id: String,
    pub symbol: String,
    pub exchange: String,
    pub reason: Option<String>,
    pub created_at: String,
    pub created_by: String,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
pub struct BlacklistCreateRequest {
    pub symbol: String,
    pub exchange: String,
    pub reason: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BlacklistListResponse {
    pub items: Vec<BlacklistItemResponse>,
    pub total: usize,
}
